### PR TITLE
_flatpak_command returns a str on python2 or python3

### LINKED ISSUE #46995: flatpak module throws exception
##### SUMMARY
This function returns a string in python2 and bytes in python3:
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/flatpak.py#L197

In https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/flatpak.py#L157
`name`  is a string and an error will be thrown when trying to compare a string to bytes.

```
TASK [pitivi : Install pitivi] ***************************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: a bytes-like object is required, not 'str'
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/home/jobevers/.ansible/tmp/ansible-tmp-1539388422.430241-248774512390951/AnsiballZ_flatpak.py\", line 113, in <module>\n    _ansiballz_main()\n  File \"/home/jobevers/.ansible/tmp/ansible-tmp-1539388422.430241-248774512390951/AnsiballZ_flatpak.py\", line 105, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/jobevers/.ansible/tmp/ansible-tmp-1539388422.430241-248774512390951/AnsiballZ_flatpak.py\", line 48, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/usr/lib/python3.5/imp.py\", line 234, in load_module\n    return load_source(name, filename, file)\n  File \"/usr/lib/python3.5/imp.py\", line 170, in load_source\n    module = _exec(spec, sys.modules[name])\n  File \"<frozen importlib._bootstrap>\", line 626, in _exec\n  File \"<frozen importlib._bootstrap_external>\", line 665, in exec_module\n  File \"<frozen importlib._bootstrap>\", line 222, in _call_with_frames_removed\n  File \"/tmp/ansible_flatpak_payload_mhlh9n27/__main__.py\", line 249, in <module>\n  File \"/tmp/ansible_flatpak_payload_mhlh9n27/__main__.py\", line 240, in main\n  File \"/tmp/ansible_flatpak_payload_mhlh9n27/__main__.py\", line 157, in flatpak_exists\nTypeError: a bytes-like object is required, not 'str'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
flatpak

##### ANSIBLE VERSION

```
ansible 2.8.0.dev0 (devel 6e2897647c) last updated 2018/10/12 19:24:51 (GMT -400)
  config file = None
  configured module search path = ['/home/jobevers/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jobevers/code/ansible/lib/ansible
  executable location = /home/jobevers/.local/bin/ansible
  python version = 3.5.2 (default, Nov 23 2017, 16:37:01) [GCC 5.4.0 20160609]
```

### PR DESCRIPTION
**Pull request by jobevers**
##### SUMMARY
Fixes https://github.com/ansible/ansible/issues/46995

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
flatpak

##### ANSIBLE VERSION

```
ansible 2.8.0.dev0 (str-flatpack-out 7469172add) last updated 2018/10/12 20:06:16 (GMT -400)
  config file = None
  configured module search path = ['/home/jobevers/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jobevers/code/ansible/lib/ansible
  executable location = /home/jobevers/.local/bin/ansible
  python version = 3.5.2 (default, Nov 23 2017, 16:37:01) [GCC 5.4.0 20160609]
```

### CODE REVIEW

File: lib/ansible/modules/packaging/os/flatpak.py [hunk 1]
```diff
@@ -203,7 +203,10 @@ def _flatpak_command(module, noop, command):
     result['stderr'] = stderr_data
     if result['rc'] != 0:
         module.fail_json(msg="Failed to execute flatpak command", **result)
-    return stdout_data
+    if isinstance(stdout_data, str):
```
Comment (Akasurde) [inline lib/ansible/modules/packaging/os/flatpak.py]: Use `to_native` from - https://github.com/ansible/ansible/blob/db304c27c75fd98b78dd0b2e23cb45a014ed6c17/lib/ansible/module_utils/_text.py#L260

**Comment by jobevers** [comment]
Done.  Also used to_native on the output of the same function in flatpak_remote, which should fix https://github.com/ansible/ansible/issues/46994

**Review by Akasurde — APPROVED** [review]
Not tested myself, but code look good to me.

**Comment by Akasurde** [comment]
Need another pair of eyes to approve.

**Comment by Dmitriusan** [comment]
@jobevers , thank you for providing a fix :+1: 
I'm facing this bug :(

**Comment by mfuchs** [comment]
I just realised that you already work on a fix, after I created PR #48049 
Yeah, I saw your bug reports but was too blind to see the fix. :smile: 

I think you are missing a `to_bytes` call though since `bytes` in Python3 does not take two parameters.

I'll gladly close my PR if you also handle the bytes-part.

**Comment by bcoca** [comment]
closed via #48049

**Pull request closed without merging**